govcd/api_vcd_versions.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"sort"
+
+	semver "github.com/hashicorp/go-version"
+
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+type VersionInfo struct {
+	Version    string `xml:"Version"`
+	LoginUrl   string `xml:"LoginUrl"`
+	Deprecated bool   `xml:"deprecated,attr,omitempty"`
+}
+
+type VersionInfos []VersionInfo
+
+type SupportedVersions struct {
+	VersionInfos `xml:"VersionInfo"`
+}
+
+// APIMaxVerIs allows to compare against maximum vCD supported API version. It will always
+// return false until the client has not done Authenticate(). Can be useful to validate
+// what vCD version is actually running.
+//
+// Format: ">= 27.0, < 32.0", ">= 30.0", "= 27.0"
+//
+// vCD version mapping to API version support https://code.vmware.com/doc/preview?id=8072
+func (vdcCli *VCDClient) APIMaxVerIs(versionConstraint string) bool {
```

Comment:
It should be `vcdCli`, not `vdcCli`.
This error is present in many methods through the library. It would be nice to fix all of them.

Reply:
Ok. It seemed confusing and now I know why. I will fix it either in the end of this PR (to avoid many altered files in PR) or in a separate one.